src/stream.rs

```diff
@@ -367,47 +391,73 @@ pub fn compress_buf(params: &CompressParams,
 pub fn compress_vec(params: &CompressParams,
```

Comment:
Ah ok, make sense to just bottom out at the "most low level" API as soon as possible for us I think. I'd be fine with just dropping this function entirely as it's somewhat niche anyway.